test/GPv2Settlement.test.ts

```diff
@@ -5,14 +5,22 @@ import { ethers, waffle } from "hardhat";
 import { domain } from "../src/ts";
 
 describe("GPv2Settlement", () => {
+  const [deployer, owner, solver] = waffle.provider.getWallets();
   let settlement: Contract;
+  let authenticator: Contract;
 
   beforeEach(async () => {
+    const GPv2SimpleAuthentication = await ethers.getContractFactory(
+      "GPv2SimpleAuthentication",
+      owner,
+    );
+    authenticator = await GPv2SimpleAuthentication.connect(owner).deploy();
```

Comment:
```suggestion
    authenticator = await GPv2SimpleAuthentication.deploy();
```

`connect` isn't needed here as you created the `ContractFactory` with a deployer already.

Reply:
Right I forgot to remove it.